File: typescript/linear_search.ts
class LinearSearch {
  public linearSearch(arr: number[], target: number): number {
    for(let i = 0; i < arr.length; i++) {
      if(arr[i] === target) {
        return i; // target found at index i
      }
    }
    return -1;
  }
}

// example usage
const numbers = [2, 5, 8, 12, 16, 23, 38, 56, 72, 91];
const search = new LinearSearch();
const index = search.linearSearch(numbers, 23);

if(index !== -1) {
  console.log('target found at index ${index}');
} else {
  console.log('target not found');
}

File: java/LinearSearch.java
// time complexity; O(n)
// space complexity: O(1)

public class LinearSearch
  {
    public static int linearSearch(int[] arr, int x)
    {
      int n = arr.length;
      for (int i = 0; i < n; i++)
        {
          if (arr[i] == x)
          {
            return i;
          }
        }
      return -1;
    }
  }
